bn_mp_set_double.c

```diff
@@ -14,18 +14,18 @@
  */
 
 #if defined(__STDC_IEC_559__) || defined(__GCC_IEC_559)
-int mp_set_double(mp_int *a, double d)
+int mp_set_double(mp_int *a, double b)
 {
    uint64_t frac;
    int exp, res;
    union {
       double   dbl;
       uint64_t bits;
    } cast;
-   cast.dbl = d;
+   cast.dbl = b;
 
-   exp = (int)(cast.bits >> 52) & 0x7FF;
-   frac = (cast.bits & ((1ULL << 52) - 1)) | (1ULL << 52);
+   exp = (unsigned)(cast.bits >> 52) & 0x7FFU;
```

Comment:
`exp` should be `unsigned` instead of `int` or not?

Reply:
`exp`must stay an `ìnt`

Reply:
àn explicit cast could help to clarify the expression:
```
   exp = (int)((unsigned)(cast.bits >> 52) & 0x7FFU);
```